edge-modules/edgehub-proxy/src/main.rs

```diff
@@ -126,17 +124,15 @@ fn run() -> Result<(), Error> {
         )
         .get_matches();
 
-    let mut tokio_runtime = tokio::runtime::current_thread::Runtime::new()?;
-    let url = Url::parse(
-        matches
-            .value_of("host")
-            .expect("no value for required HOST"),
-    )?;
-    let client = client(&url)?;
-
-    let api_version = matches
-        .value_of("apiversion")
-        .expect("no default value for API_VERSION");
+    let mut tokio_runtime = tokio::runtime::Builder::new_current_thread()
```

Comment:
You should be able to use #[tokio::main] instead of manually making the runtume: https://docs.rs/tokio/1.10.1/tokio/attr.main.html

Reply:
I see. Handy change. Thanks

Reply:
In line 158 of the same file, how would I replicate the

let response = tokio_runtime.block_on(request)?;

line? Or a better question: how do I define the runtimes entrypoint using this macro instead of manually creating it

Reply:
> In line 158 of the same file, how would I replicate the

Since `request` is just a future, you should be able to just `.await` that, right?

> Or a better question: how do I define the runtimes entrypoint using this macro instead of manually creating it

You just put `#[tokio::main]` on `async fn main()`, also making `main` and `run` both async fn.